crawl-ref/source/dat/des/variable/float.des

```diff
@@ -4901,6 +5115,148 @@ MAP
 .cccccccccc
 ENDMAP
 
+# If you keep messing around with terrain-based magic in the same area,
+# sometimes things get messy. Monster set focuses on monsters that modify
+# terrain (Primal Wave, Eruption, Hurl Sludge, Dig, water aura, etc.) and/or use
+# it tactically (Waterstrike, Awaken Forest, Pyroclastic Surge, LRD, etc.).
+NAME:     nicolae_terramorphic_expanse
+TAGS:     patrolling no_pool_fixup transparent
+DEPTH:    D:11-, Depths
+ORIENT:   float
+KITEM:    1% = % w:30 / * / | w:5
+KMONS:    1 = salamander tyrant / merfolk aquamancer / water nymph / \
+              fenstrider witch / spriggan druid / eleionoma / \
+              deep elf elementalist / deep troll earth mage
```

Comment:
Some comments on monster difficulty. In terms of acceptability, t's helpful to operate off of a common ground rule like "anything up to an `8` is acceptable, but never worse than this, and the number of such super-out-of-depth spawns should be reasonable". Here a monster's depth would be based on the branch's OOD table or the earliest place where the monster naturally generates. This prevents rules about what's acceptable from being very individual to the reviewer. There's still going to be subjectivity about what's a reasonable number of such spawns or when a vault should have a runed door when something is so extreme it should have a rune door, etc. But at least in terms of whether a monster is even acceptable, basically by D:13- you can at the very top end potentially go to any naturally spawning monster in the game. For the most extreme monsters, you would want a runed door and to have spawn counts very low (this monster type is the "boss"). Deep troll earth mages are actually in the Dungeon OOD table, so it's certainly fine to place those in Dungeon, and I don't think any monster on that list needs to be excluded from Dungeon; in late Dungeon they are basically just `9` spawns, since they are effectively spawn depth 16-18.

It might work best to condition placement for D:11-13, say, placing 1 to 2 instances (or just 1, if you're nice) of the top end threat on the `1` list , and allow, say, 2-4 (or even all 4!) to place for D:14-. From what I can see, only 4 place at max, but I haven't examined the subvaults. These are nasty spawns and it's an open vault, but I think it's fine to place these monsters if you control the counts carefully by depth.

Reply:
> From what I can see, only 4 place at max, but I haven't examined the subvaults. 

nicolae_terramorphic_biome is where most monsters get placed. I did the math; in addition to the four monsters placed in the main vault, the average is 19.2 monsters per vault, although this doesn't include the "nothing" options included in the KMONS statements in D. 

Conditioning on depth+branch might work, but it might just be simpler to make it Depths-only. I'll think about it some more.

Reply:
terramorphic_expanse is Depths only, and I've tweaked the monster distribution somewhat.